beacon-chain/sync/rpc_status.go

```diff
@@ -15,33 +15,46 @@ import (
 	"github.com/prysmaticlabs/prysm/shared/roughtime"
 	"github.com/prysmaticlabs/prysm/shared/runutil"
 	"github.com/prysmaticlabs/prysm/shared/slotutil"
+	"github.com/sirupsen/logrus"
 )
 
-const statusInterval = 6 * time.Minute // 30 slots.
-
 // maintainPeerStatuses by infrequently polling peers for their latest status.
 func (r *Service) maintainPeerStatuses() {
-	ctx := context.Background()
-	runutil.RunEvery(r.ctx, statusInterval, func() {
+	// Run twice per epoch.
+	interval := time.Duration(params.BeaconConfig().SecondsPerSlot*params.BeaconConfig().SlotsPerEpoch/2) * time.Second
+	runutil.RunEvery(r.ctx, interval, func() {
 		for _, pid := range r.p2p.Peers().Connected() {
 			// If the status hasn't been updated in the recent interval time.
 			lastUpdated, err := r.p2p.Peers().ChainStateLastUpdated(pid)
 			if err != nil {
 				// Peer has vanished; nothing to do
 				continue
 			}
-			if roughtime.Now().After(lastUpdated.Add(statusInterval)) {
-				if err := r.sendRPCStatusRequest(ctx, pid); err != nil {
+			if roughtime.Now().After(lastUpdated.Add(interval)) {
+				if err := r.sendRPCStatusRequest(r.ctx, pid); err != nil {
```

Comment:
for each request I dont think we should be sending in the parent service's context.

Reply:
The idea of passing down the service's context is if it is cancelled then the downstream call has a chance to notice it.  If we just passed `context.Background()` (or even a cancellable context that was newly minted in this function) we'd lose this functionality.

Reply:
If this service's context was canceled then we would most likely be disconnected with the peer by the time they receive. Maybe we could create a context with a span ? This would also help us in tracing